spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -644,28 +646,43 @@ def add_electrical_series(recording: se.RecordingExtractor, nwbfile=None, metada
                 scalar_conversion = 1.
                 channel_conversion = gains * 1e-6
 
-            def data_generator(recording, channels_ids):
-                #  generates data chunks for iterator
-                for id in channels_ids:
-                    data = recording.get_traces(channel_ids=[id]).flatten()
-                    yield data
-
-            data = data_generator(recording=recording, channels_ids=channel_ids)
-            ephys_data = DataChunkIterator(data=data, iter_axis=1)
-
-            # To get traces in Volts = data*channel_conversion*conversion
-            ephys_ts = ElectricalSeries(
-                name=es_name,
-                data=ephys_data,
-                electrodes=electrode_table_region,
-                starting_time=recording.frame_to_time(0),
-                rate=recording.get_sampling_frequency(),
-                conversion=scalar_conversion,
-                channel_conversion=channel_conversion,
-                comments='Generated from SpikeInterface::NwbRecordingExtractor',
-                description=metadata['Ecephys']['ElectricalSeries'].get('description', defaults['description'])
+            if isinstance(recording.get_traces(), np.memmap):
+                ephys_data = H5DataIO(
+                    DataChunkIterator(
+                        data=np.memmap.transpose(recording.get_traces()),  # nwb standard is time as zero axis
+                        buffer_size=10**6
```

Comment:
@bendichter @alejoe91 Setting the buffer_size is essentially like setting the chunk size in the data generator below. Too low and the process will take much longer to run than otherwise necessary; too high and it _may_ trigger memory problems.

The 10**6 is currently hardcoded based on my test runs on a remote server; I did some minimal testing locally but don't know how well it scales. Any ideas for better setting this value or determining it from the system calling it?

Reply:
Yes :) 
Check this: https://github.com/SpikeInterface/spikeextractors/blob/master/spikeextractors/extraction_tools.py#L269-L316

Here the chunk size is either passed woth the `chunk_size`arg (default `None`) or it's computed based on how much RAM you allow the writing to use (`chunk_mb=500`). I think having something like this could be nice.

Reply:
Yeah, I remember that function; and actually being able to specific RAM overhead is a really nice feature but haven't looked into how to actually do that. Will do that now.

Reply:
@alejoe91 OK, check out that latest commit and comments.